spring-test/src/main/java/org/springframework/test/context/event/AfterClassTestExecutionEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.context.event;
+
+import org.springframework.test.context.TestContext;
+
+/**
+ * {@link TestExecutionEvent} published by {@link EventPublishingTestExecutionListener} when
+ * {@link org.springframework.test.context.TestExecutionListener#afterTestClass(TestContext)} is called.
+ *
+ * @author Frank Scheffler
+ * @since 5.2
+ */
+@SuppressWarnings("serial")
+public class AfterClassTestExecutionEvent extends TestExecutionEvent {
```

Comment:
Let's rename all of the event types so that there is a one-to-one relationship between the names of the callbacks in TestExecutionListener. This will make things more straightforward and future-proof (in case additional, similar methods are introduced later).

- `BeforeTestClassEvent`
- `BeforeTestExecutionEvent`
- `BeforeTestMethodEvent`
- `PrepareTestInstanceEvent`
- `AfterTestMethodEvent`
- `AfterTestExecutionEvent`
- `AfterTestClassEvent`

Reply:
Any suggestions regarding the name for the base class, now: TestExecutionEvent?

Reply:
How about `TestContextEvent` since they are events fired from the _Spring TestContext Framework_? 🤔

Reply:
fine for me.